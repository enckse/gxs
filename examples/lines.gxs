# comments start with '#' but can NOT be within a '{' and '}' block
palette => {
    x => red
    y => #333333
    z => NONE
}
mode => {
    hline
}
pattern => {
    xyzyz
    zyxyz
    zzxzz
}
action => {
    commit
}
# reuse the palette
mode => {
    vline
}
# patterns must always be redefined
pattern => {
    xxxxx
    zzzzz
    xxxxx
}
action => {
    commit
}
palette => {
    x => orange
    y => pink
    z => NONE
}
mode => {
    xstitch
}
pattern => {
    xyzyz
    zyxyz
    zzxzz
}
action => {
    commit
}
