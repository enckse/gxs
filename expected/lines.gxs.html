<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<style>
.container {
  background: white;
  display: inline-block;
  border: 1px dotted black;
}
.grid {
  display: grid;
  grid-template-columns: repeat(6, 10px);
  grid-template-rows: repeat(6, 10px);
  grid-gap: 1px;
}
.cell {
  justify-content: center;
  align-items: center;
  display: flex;
  font-family: Arial;
  font-size: 4pt;
  font-weight: bold;
  background: white;
}
.main {
  margin-left: 10px;
  padding: 0px 10px;
}
</style>
    </head>
    <body>
        <div class="main">
  <div class="container">
  <div class="grid" id="grid">
      <div class="cell" style="" id="000x000"></div>
      <div class="cell" style="" id="001x000">1</div>
      <div class="cell" style="" id="002x000">2</div>
      <div class="cell" style="" id="003x000">3</div>
      <div class="cell" style="" id="004x000">4</div>
      <div class="cell" style="" id="005x000">5</div>
      <div class="cell" style="" id="000x001">1</div>
      <div class="cell" style="color:  rgb(199, 43, 59);color:  rgb(199, 43, 59);background-color:  orange" id="001x001">--|--</div>
      <div class="cell" style="color:  #333333;color:  rgb(199, 43, 59);background-color:  pink" id="002x001">--|--</div>
      <div class="cell" style="color:  rgb(199, 43, 59)" id="003x001">|</div>
      <div class="cell" style="color:  #333333;color:  rgb(199, 43, 59);background-color:  pink" id="004x001">--|--</div>
      <div class="cell" style="color:  rgb(199, 43, 59)" id="005x001">|</div>
      <div class="cell" style="" id="000x002">2</div>
      <div class="cell" style="" id="001x002"></div>
      <div class="cell" style="color:  #333333;background-color:  pink" id="002x002">-----</div>
      <div class="cell" style="color:  rgb(199, 43, 59);background-color:  orange" id="003x002">-----</div>
      <div class="cell" style="color:  #333333;background-color:  pink" id="004x002">-----</div>
      <div class="cell" style="" id="005x002"></div>
      <div class="cell" style="" id="000x003">3</div>
      <div class="cell" style="color:  rgb(199, 43, 59)" id="001x003">|</div>
      <div class="cell" style="color:  rgb(199, 43, 59)" id="002x003">|</div>
      <div class="cell" style="color:  rgb(199, 43, 59);color:  rgb(199, 43, 59);background-color:  orange" id="003x003">--|--</div>
      <div class="cell" style="color:  rgb(199, 43, 59)" id="004x003">|</div>
      <div class="cell" style="color:  rgb(199, 43, 59)" id="005x003">|</div>
      <div class="cell" style="" id="000x004">4</div>
      <div class="cell" style="" id="001x004"></div>
      <div class="cell" style="" id="002x004"></div>
      <div class="cell" style="" id="003x004"></div>
      <div class="cell" style="" id="004x004"></div>
      <div class="cell" style="" id="005x004"></div>
      <div class="cell" style="" id="000x005">5</div>
      <div class="cell" style="" id="001x005"></div>
      <div class="cell" style="" id="002x005"></div>
      <div class="cell" style="" id="003x005"></div>
      <div class="cell" style="" id="004x005"></div>
      <div class="cell" style="" id="005x005"></div>
  </div>
</div>
        </div>
    </body>
</html>
